src/wp-includes/block-supports/shadow.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Shadow block support flag.
+ *
+ * @package WordPress
+ * @since 6.3.0
+ */
+
+/**
+ * Registers the style and shadow block attributes for block types that support it.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function wp_register_shadow_support( $block_type ) {
+	$has_shadow_support = block_has_support( $block_type, array( 'shadow' ), false );
+
+	if ( ! $has_shadow_support ) {
+		return;
+	}
+
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	if ( $has_shadow_support && ! array_key_exists( 'style', $block_type->attributes ) ) {
```

Comment:
```suggestion
	if ( ! array_key_exists( 'style', $block_type->attributes ) ) {
```

We have already checked for block support above, and this code will run if it is `true`.